agent_patch_distr.py

```diff
@@ -0,0 +1,245 @@
+            y_values = np.append(y_values, posy[j,i,nr_of_radii - 1 - i,:,0][:int(number_of_entities)] - window_pad + int(radius_of_entity))
+
+        # making sure that no drawn location is outside the arena borders
+        max_x_values = np.max(x_values)
+        max_y_values = np.max(y_values)
+        if (max_x_values > ENV_WIDTH) | (max_y_values > ENV_HEIGHT):
+            sys.exit('At least one patch location is outside the arena borders.')
+
+        # loop through agent/patch locations and "create" a filled CIRCLE for each location in pixel array
+        pixel_array = np.zeros([ENV_HEIGHT,ENV_WIDTH])
+        for k in range(0,len(x_values)):
+            pixel_array = modifiy_pixel_array_circle(pixel_array, x_values[k], y_values[k], radius_of_entity)
+        # normalized_pixel_array = pixel_array / pixel_array.sum() # this is the probability density for num_batches initializations
+        normalized_pixel_array = pixel_array / num_batches # this is the frequency a pixel is covered
+        # save pixel array for adapting plots (computation of pixel array is slow)
+        np.savez(folderpath +'/{}_normalized_pixel_array_circles_R_{}_N_{}_R_R_{}_N_R_{}_batches_{}.npz'.format(agent_or_patch,round(RADIUS_AGENT,2),int(N),round(RADIUS_RESOURCE[nr_of_radii - 1 - i],2),int(N_RESOURCES[i]),num_batches),normalized_pixel_array=normalized_pixel_array)
+        plot_circles(agent_or_patch, normalized_pixel_array, RADIUS_AGENT, N, RADIUS_RESOURCE[nr_of_radii - 1 - i], N_RESOURCES[i], num_batches, folderpath)
+
+        # loop through agent/patch locations and add + 1 for each PIXEL location in pixel array
+        pixel_array = np.zeros([ENV_HEIGHT,ENV_WIDTH])
+        for k in range(0,len(x_values)):
+            pixel_array[int(y_values[k]),int(x_values[k])] = pixel_array[int(y_values[k]),int(x_values[k])] + 1
+        # normalized_pixel_array = pixel_array / pixel_array.sum() # this is the probability density for num_batches initializations
+        normalized_pixel_array = pixel_array / num_batches # this is the frequency a pixel is covered
+        # save pixel array for adapting plots (computation of pixel array is slow)
+        np.savez(folderpath +'/{}_normalized_pixel_array_locations_R_{}_N_{}_R_R_{}_N_R_{}_batches_{}.npz'.format(agent_or_patch,round(RADIUS_AGENT,2),int(N),round(RADIUS_RESOURCE[nr_of_radii - 1 - i],2),int(N_RESOURCES[i]),num_batches),normalized_pixel_array=normalized_pixel_array)
+        plot_centers(agent_or_patch,normalized_pixel_array, RADIUS_AGENT, N, RADIUS_RESOURCE[nr_of_radii - 1 - i], N_RESOURCES[i], num_batches, folderpath)
+    return
+
+def loop_batches_small_patch(agent_or_patch, folderpath, ENV_HEIGHT, ENV_WIDTH, posx, posy, RADIUS_AGENT, N, N_RESOURCES, RADIUS_RESOURCE, num_batches, window_pad):
+    x_values = []
+    y_values = []
+    for j in range(0,num_batches):
```

Comment:
This again can be done with 2 lines of proper numpy indexing instead of for loops